### PR TITLE
validate_batch() and train_batch() issues

### PR DESCRIPTION
**Pull request by leonardozcm**
fix `validate_batch()` and `train_batch() `issues: 
1. target size does not match the size of output when doing criterion in validate_batch(). 
2. loss is not averaged?

### CODE REVIEW

File: pyzoo/zoo/orca/learn/pytorch/training_operator.py [hunk 1]
```diff
@@ -349,10 +349,12 @@ def validate_batch(self, batch, batch_info):
...
-        if len(target.size()) > 1:
-            # Can't directly call torch.squeeze() in case batch size is 1.
-            for i in reversed(range(1, len(target.size()))):
-                target = torch.squeeze(target, i)
+        # if len(target.size()) > 1:
+        #     # Can't directly call torch.squeeze() in case batch size is 1.
+        #     print(range(1,len(target.size())))
+        #     for i in reversed(range(1, len(target.size()))):
+        #         target = torch.squeeze(target, i)
+        #         print(target.size())
```
Comment (hkvision) [inline pyzoo/zoo/orca/learn/pytorch/training_operator.py:357]: Do not comment out these lines. Put them after calculating the loss instead.

File: pyzoo/zoo/orca/learn/pytorch/training_operator.py [hunk 2]
```diff
@@ -388,8 +390,9 @@ def validate_batch(self, batch, batch_info):
 
         num_correct = np.sum(np_output == np_target)
         num_samples = target.size(0)
+        # print("val_size: "+str(num_samples))
```
Comment (hkvision) [inline pyzoo/zoo/orca/learn/pytorch/training_operator.py:393]: remove

File: pyzoo/zoo/orca/learn/pytorch/training_operator.py [hunk 3]
```diff
@@ -361,13 +358,18 @@ def validate_batch(self, batch, batch_info):
...
 
         # compute output
-        with self.timers.record("eval_fwd"):
-            output = self.model(*features)
-            loss = self.criterion(output, target)
-            if len(output.size()) > 1:
-                # In case there is extra trailing dimensions.
-                for i in reversed(range(1, len(output.size()))):
-                    output = torch.squeeze(output, i)
+            with self.timers.record("eval_fwd"):
```
Comment (hkvision) [inline pyzoo/zoo/orca/learn/pytorch/training_operator.py:361]: why change the indentation?

File: pyzoo/zoo/orca/learn/pytorch/training_operator.py [hunk 4]
```diff
@@ -349,10 +349,7 @@ def validate_batch(self, batch, batch_info):
         """
         # unpack features into list to support multiple inputs model
         *features, target = batch
-        if len(target.size()) > 1:
-            # Can't directly call torch.squeeze() in case batch size is 1.
-            for i in reversed(range(1, len(target.size()))):
-                target = torch.squeeze(target, i)
+
```
Comment (hkvision) [inline pyzoo/zoo/orca/learn/pytorch/training_operator.py:352]: remove this line

**Comment by hkvision** [comment]
Make this two separate PRs. I will merge the target squeeze part for you first.

File: pyzoo/zoo/orca/learn/pytorch/training_operator.py [hunk 5]
```diff
@@ -364,6 +360,11 @@ def validate_batch(self, batch, batch_info):
         with self.timers.record("eval_fwd"):
             output = self.model(*features)
             loss = self.criterion(output, target)
+            if len(target.size()) > 1:
+                # Can't directly call torch.squeeze() in case batch size is 1.
+                for i in reversed(range(1, len(target.size()))):
+                    target = torch.squeeze(target, i)
+                    print(target.size())
```
Comment (hkvision) [inline pyzoo/zoo/orca/learn/pytorch/training_operator.py:367]: remove print

**Review by hkvision — APPROVED** [review]

**Comment by leonardozcm** [comment]
jenkins passed [https://github.com/intel-analytics/analytics-zoo/pull/3033](https://github.com/intel-analytics/analytics-zoo/pull/3033)

**Comment by hkvision** [comment]
http://10.239.47.210:18888/view/ZOO-PR/job/ZOO-PR-Validation/4488/
